### ISSUE TITLE
[F#] Implement new Concept Exercise: operator-overloading

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#857]
This issue describes how to implement the `operator-overloading` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Operator Overloading in F#.

## Learning objectives

- Know that operators are regular functions.
- Know how to add custom operators for built-in types.
- Know how to add custom operators for user-defined types.
- Know how to overload existing operators for built-in types.
- Know how to overload existing operators for user-defined types.
- Know how to define prefix and infix operators.

## Out of scope

Nothing.

## Concepts

The Concepts this exercise unlocks are:

- `operator-overloading`: know that operators are regular functions; know how to add custom operators for built-in types; know how to add custom operators for user-defined types; know how to overload existing operators for built-in types; know how to overload existing operators for user-defined types; know how to define prefix and infix operators.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `records`: to define an operator that work on records.
- `classes`: to define an operator that work on classes.
- `static-members`: know how to define static members on classes.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Operator overloading syntax][operator-overloading]: basic syntax for operator overloading.
- [Allowed operators][allowed-operators]: lists the operators allowed to overload.
- [Prefix and infix operators][prefix-and-infix-operators]: how to define prefix or infix operators.
- [Global operators][global-operators]: how to define global operators.
- [Type-specific operators][type-operators]: how to define type-specific operators.

### After

- [Operator overloading syntax][operator-overloading]: basic syntax for operator overloading.
- [Allowed operators][allowed-operators]: lists the operators allowed to overload.
- [Prefix and infix operators][prefix-and-infix-operators]: how to define prefix or infix operators.
- [Global operators][global-operators]: how to define global operators.
- [Type-specific operators][type-operators]: how to define type-specific operators.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/concepts/operators.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[operator-overloading]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/operator-overloading#syntax
[allowed-operators]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/operator-overloading#creating-new-operators
[prefix-and-infix-operators]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/operator-overloading#prefix-and-infix-operators
[global-operators]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/operator-overloading#operators-at-the-global-level
[type-operators]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/operator-overloading#remarks
